# Config file for the Auto Mapping ROS node
package_name: "auto_mapping_ros"

# basic config
num_vehicles: 1

csv_filepath: "/csv/sequence"
csv_localtraj: "/csv/local_traj_50_fwd_rev.csv"

# Local planner option
#   "pure_pursuit" or "mpc"
local_planner_mode: "pure_pursuit"

# Parameter for pure pursuit algorithm
# A good estimate is 0-2r (According to the implementation explanation given in RI, CMU)
lookahead_distance: 1.25

# Whether waypoints need non-ROS to ROS coordinate transformation
translate_non_ros_coord: false

# Non ROS Map Values
# (User needs to change these values when changing maps)
non_ros_map_width: 1 # map: world coordinates defined by Revit
non_ros_map_height: 1
switch_xy: false

# Distance threshold to start planning for the next node
distance_threshold: 1.5 # map: levine_4_cad

# Topics to publish and subscribe to
pose_topic: "/pf/viz/inferred_pose"
drive_topic: "/vesc/high_level/ackermann_cmd_mux/input/nav_1"
brake_topic: "brake_bool"
scan_topic: "scan"

# Frame Names
base_frame: "laser"
map_frame: "map"

# Global Planner Name
planner_name: "fmt_star_server"

# Velocity of the cars
velocity: 0.8

# Local planner (MPC)

slip_p1_vel: 5.2
slip_p1_steer: 1.41
slip_p2_vel: 5.5
slip_p2_steer: 1.36

follow_gap_thresh: 3
state_lims: 1
fov_divider: 1.5
buffer: 3

q0: 10000.0
q1: 10000.0
q2: 0.0

r0: 0.00001
r1: 0.000

horizon: 30
dt: 0.05
#occ_grid
occ_size: 10
occ_discrete: 0.1
occ_dilation: 0.05
cmaes_lookahead_1: -1.0
cmaes_lookahead_2: 2.0
switch_distance_threshold: 3.0

#traj_planner
num_traj: 20
close_weight: 0.5
MAX_HORIZON: 50
rev_threshold: 1

# Lanes
lane_0: "csv/temp.csv"

# desired input
des_vel: 0.3
des_steer: 0
#constraints
umax: 0.5 #change dt if you change this
umin: -0.5
